src/main/java/com/thealgorithms/misc/ShuffleArray.java

```diff
@@ -0,0 +1,73 @@
+package com.thealgorithms.misc;
+import java.util.Random;
+import java.util.Scanner;
+/**
+ * The Fisher-Yates (Knuth) Shuffle algorithm randomly permutes an array's
+ * elements, ensuring each permutation is equally likely.
+ *
+ * <p>
+ * Worst-case performance O(n)
+ * Best-case performance O(n)
+ * Average performance O(n)
+ * Worst-case space complexity O(1)
+ *
+ * This class provides a static method to shuffle an array in place.
+ *
+ * @author Your Name (https://github.com/Chiefpatwal)
+ */
+public final class ShuffleArray {
+    // Prevent instantiation
+    private ShuffleArray() {
+    }
+    /**
+     * This method shuffles an array using the Fisher-Yates algorithm.
+     *
+     * @param arr is the input array to be shuffled
+     */
+    public static void shuffle(int[] arr) {
+        Random random = new Random();
+        for (int i = arr.length - 1; i > 0; i--) {
+            int j = random.nextInt(i + 1);
+            int temp = arr[i];
+            arr[i] = arr[j];
+            arr[j] = temp;
+        }
+    }
+    /**
+     * This method takes user input to create an array.
+     *
+     * @return the array created from user input
+     */
+    public static int[] getUserInputArray() {
```

Comment:
Please replace user input and main with JUnit tests

Reply:
yes working on it!

Reply:
> Please replace user input and main with JUnit tests

@siriak please check :) 
Thanks!!